### ISSUE TITLE
jQuery.detach() Pattern May Yield Unexpected Results

### ISSUE THREAD

**Issue by zikes** [shichuan/javascript-patterns#29]
Currently the pattern for detaching an element from the DOM before manipulating and reinserting it does not account for its position within the parent element.  In the example given at https://github.com/shichuan/javascript-patterns/blob/master/jquery-patterns/detach.html if the table were the first child element under the parent it would end up as the last child element once the operation completed.
